### ISSUE TITLE
Out-of-dashboard keyboard is always at full size

### ISSUE THREAD

**Issue by matejdro** [elvissteinjr/DesktopPlus#37]
For me, it seems like keyboard that appears on overlays that are outside Steam dashboard, is always at maximum size, regardless of the setting.

https://user-images.githubusercontent.com/507922/147680478-d9c157ac-e03b-4937-803a-01d7986c03df.mp4

Running NewUI Preview 4

**Comment by elvissteinjr** [comment]
The out-of-dashboard-tab keyboard size is separate from the in-dashboard-tab one. The settings window displays the currently active one, meaning the settings window needs to be pinned and displayed outside of the dashboard tab to change it there.
The idea was that in most cases the keyboard window is resized from motion controller input (horizontal stick movement while dragging or two-handed gesture drag by holding right click on of the controllers).

I suppose the current state of things is more awkward than just listing both sizes, though. The other two UI windows with persistent state could probably also benefit from having some of their values more accessible in this manner.
I'll see how exactly to handle this, but I'm also open to suggestions here.

**Comment by matejdro** [comment]
That worked, thanks, but I think design here is a bit unintuitive. All other settings are global, but keyboard is somehow separate for the out-of-dashboard and for the in-dashboard.

Maybe moving the keyboard size to the overlay properties window would make more sense?

**Comment by elvissteinjr** [comment]
Not sure if Overlay Properties is a good fit as the state is still global, but split into the two states, not specific to the overlay the properties window is displaying. Same applies to the other two UI windows, so it's not just the keyboard, but I guess it's less apparent there as there's no size option exposed in the first place.

Haven't started working on that yet, but my current idea would be a little sub-page where you can modify and reset the UI state parameters for each window directly. My only worry there is if it wouldn't be a bit too in-depth.

Just not having separate sizes wouldn't have worked for the keyboard though in my opinion. Size requirements between the the dashboard and overlay-attached could vary wildly. Won't deny the keyboard size UI solution being a bit rushed out in the current build, however.

**Comment by matejdro** [comment]
> Haven't started working on that yet, but my current idea would be a little sub-page where you can modify and reset the UI state parameters for each window directly.

Yeah that would make sense. I'm the king of the "too in-depth" though, so I don't think I'm the right person to discuss this.

> Won't deny the keyboard size UI solution being a bit rushed out in the current build, however.

Totally no problems here from my side. It is called preview build after all and I understand it.

**Comment by elvissteinjr** [comment]
I'm cleaning up some old, still open, issues.
The Persistent UI Settings page was added back in NewUI Preview 5, so I assume this can be considered resolved?
I'm at least not currently looking to change keyboard scaling behavior further from how it is right now.

If not this can be re-opened, of course.
